cmd/main.go

```diff
@@ -189,64 +189,37 @@ func watchConfigFile(filename, adapterName string) {
 			With(zap.String("config_file", filename))
 	}
 
-	// get the initial timestamp on the config file
-	info, err := os.Stat(filename)
+	// get current config
+	lastCfg, _, err := LoadConfig(filename, adapterName)
 	if err != nil {
-		logger().Error("cannot watch config file", zap.Error(err))
+		logger().Error("unable to load latest config", zap.Error(err))
 		return
 	}
-	lastModified := info.ModTime()
 
 	logger().Info("watching config file for changes")
 
-	// if the file disappears or something, we can
-	// stop polling if the error lasts long enough
-	var lastErr time.Time
-	finalError := func(err error) bool {
-		if lastErr.IsZero() {
-			lastErr = time.Now()
-			return false
-		}
-		if time.Since(lastErr) > 30*time.Second {
-			logger().Error("giving up watching config file; too many errors",
-				zap.Error(err))
-			return true
-		}
-		return false
-	}
-
 	// begin poller
 	//nolint:staticcheck
-	for range time.Tick(1 * time.Second) {
-		// get the file info
-		info, err := os.Stat(filename)
+	for range time.Tick(3 * time.Second) {
```

Comment:
Is this to alleviate CPU and I/O load for large configs?

I wonder if 1-2s is still fast enough for most people.

Reply:
I wonder this myself, Do you have maybe a large config file to test this approach?